buildtest/buildsystem/schemas/script/script-v0.0.1.schema.json

```diff
@@ -3,7 +3,7 @@
   "$schema": "http://json-schema.org/draft-07/schema#",
   "title": "BuildTest Test Configuration for Script",
   "type": "object",
-  "required": ["type", "run"],
+  "required": ["type", "executor","run"],
```

Comment:
This is breaking portability. A test recipe should not have a hard coded executor, it’s determined by the runners config or command line specification at runtime. Also, changes to the schema don’t go here, they are reviewed first in the schemas repository.

Reply:
i would prefer having ``executor`` in recipe otherwise it will be problematic especially when it comes to doing directory build (`buildtest build -c dir``) where more than one yml file is processed that may use two different executor. For example, there is one executor for GPU partition, and another for MPI. I have no way of doing this via cli other than running two separate builds. 

Sure, i will make change to same change to schema repo as well.

Reply:
This would be akin to hardcoding an executor into a Snakefile, meaning that if I move my Snakemake to some other environment I can't just run it as is. The entire purpose of having the user define executors for their host and then specify to use them either with default config or a flag is to make these recipes portable. If executor is added, it should be an *optional* value and then the client here needs to have a flag to ignore it so the recipes are still portable.

Reply:
I understand, in that case i would be fine with making executor as an optional argument and have default executor be one of the local executors. I can make that change

Reply:
The default executor should be whatever the use sets it to be in their local configuration (which I agree would be a local executor when first installed).

Reply:
The change has been made in https://github.com/HPC-buildtest/buildtest-framework/pull/249/commits/a5ab2e7d1f47c99fa936c646c4051f4280964b71 so executor is no longer required. I fixed the Travis issue so it should be all green now.